packages/l/libkmod/xmake.lua

```diff
@@ -6,18 +6,130 @@ package("libkmod")
+        -- bionic, fread_unlocked
+        if package:version():ge("v34") then
+            local ndk_sdkver = package:toolchain("ndk"):config("ndk_sdkver")
+            assert(ndk_sdkver and tonumber(ndk_sdkver) >= 28, "package(libkmod): require ndk api level >= 28")
+        end
+    end)
+
+    on_load(function (package)
+        if package:version():lt("v34") then
+            package:add("deps", "autotools")
+        else
+            package:add("deps", "meson", "ninja")
+        end
+
+        for _, lib in ipairs({"zstd", "zlib", "xz"}) do
+            if package:config(lib) then
+                package:add("deps", lib)
+            end
+        end
+        local openssl = package:config("openssl")
+        if openssl then
+            package:add("deps", openssl)
+        end
+    end)
+
+    on_install("linux", "android@linux,macosx", function (package)
+        if package:is_plat("android") then
+            local ndk_sdkver = package:toolchain("ndk"):config("ndk_sdkver")
+            if package:version():lt("v33") and tonumber(ndk_sdkver) <= 23 then
+                io.replace("shared/util.h", "#include <time.h>", [[
+                    #include <time.h>
+                    #include <string.h>
+
+                    // from libkmod v33
+                    static inline const char *basename(const char *s)
+                    {
+                        const char *p = strrchr(s, '/');
+                        return p ? p + 1 : s;
+                    }
+                ]], {plain = true})
+            end
+            io.replace("shared/util.h", "#include <time.h>", [[
+                #include <time.h>
+                #include <unistd.h>
+
+                // from https://android.googlesource.com/kernel/common/+/03c04a7cba972/tools/perf/util/get_current_dir_name.c
+                static inline char *get_current_dir_name(void)
+                {
+                    char pwd[PATH_MAX];
+                    return getcwd(pwd, sizeof(pwd)) == NULL ? NULL : strdup(pwd);
+                }
+            ]], {plain = true})
+        end
+
+        if package:version():lt("v34") then
+            local configs = {
+                "--disable-dependency-tracking",
+                "--disable-manpages",
+                "--disable-test-modules",
+                "--disable-tools"
+            }
+
+            table.insert(configs, "--enable-logging=" .. (package:config("logging") and "yes" or "no"))
+            table.insert(configs, "--with-zstd=" .. (package:config("zstd") and "yes" or "no"))
+            table.insert(configs, "--with-zlib=" .. (package:config("zlib") and "yes" or "no"))
+            table.insert(configs, "--with-xz=" .. (package:config("xz") and "yes" or "no"))
+            table.insert(configs, "--with-openssl=" .. (package:config("openssl") and "yes" or "no"))
+
+            local packagedeps = {}
+            for _, dep in ipairs(package:librarydeps()) do
+                table.insert(packagedeps, dep:name())
```

Comment:
All dependencies cannot be found? Setting packagedeps is only necessary if a dependency cannot be found automatically by pkg-config/cmake findpackage.